demo/mobile/src/app/sync/syncGlobalManager.js

```diff
@@ -44,7 +44,9 @@ SyncManager.prototype.manageDataset = function(datasetId, options, queryParams,
 SyncManager.prototype.removeManagers = function() {
   if (this.syncManagers) {
     this.syncManagers.forEach(function(syncDatasetManager) {
-      syncDatasetManager.safeStop(); //stop sync for this dataset
+      syncDatasetManager.safeStop().then(function() {
+        syncDatasetManager.clearCache(); //clear cache for this dataset
+      }); //stop sync for this dataset
```

Comment:
optional, but we can take this opportunity to make this method actually return a promise:
```javascript
return Bluebird.map(this.syncManagers, m => m.safeStop()
  .then(() => m.clearCache())
  .then(() => this.syncManagers = []))
```
Allows client code to wait for finish and catch errors.